great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -3078,13 +3073,13 @@ def _instantiate_datasource_from_config_and_update_project_config(
                 )
                 self._cached_datasources[config.name] = datasource
             except ge_exceptions.DatasourceInitializationError as e:
-                # Do not keep configuration that could not be instantiated.
-                if save_changes:
-                    self._datasource_store.delete(config)  # type: ignore[attr-defined]
-                # If the DatasourceStore uses an InlineStoreBackend, the config may already be updated
-                self.config.datasources.pop(config.name, None)  # type: ignore[union-attr,arg-type]
                 raise e
```

Comment:
Does leaving this `except` in serve any purpose now? It seems like we can remove the entire `try/except` here.